java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/MainActivity.java

```diff
@@ -168,64 +153,58 @@ private void doConnect() {
 			settings.setIsMultiInstanceAware(true);
 			settings.setIsMRA(true);
 			mService.start(MainActivity.this, MainActivity.this);
-    	}
-		else {
+    	} else {
 			Toast.makeText(MainActivity.this, "App is already connected", Toast.LENGTH_LONG).show();
 		}
 	}
 	
 	public void onListItemClick(ListView parent, View v, int position, long id) {
-        TextView item = (TextView) v;
-        Intent intent = null;
-        if (mService.isAppConnected()) {
-	        switch(position) {
-	        case 0: 
-        		intent = new Intent(MainActivity.this, WeightActivity.class);
-	        }
-        }
-        
-        if(intent != null) {
-        	startActivity(intent);
-        }
-    }
+		TextView item = (TextView) v;
+		Intent intent = null;
+		if (mService.isAppConnected()) {
+			switch(position) {
+				case 0:
+					intent = new Intent(MainActivity.this, WeightActivity.class);
```

Comment:
add "break;" to the case to avoid potential for a fallthrough case in the future if someone else ads a new case.